src/Multicall4.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+import { Multicall3 } from "./Multicall3.sol";
+
+/// @title Multicall4
+/// @notice Extension of Multicall3 with additional hooks to drain non-deterministally obtained tokens during execution
+contract Multicall4 is Multicall3 {
+    /// @notice Send the full ether balance to the sender
+    /// @dev Used to sweep any non-deterministically obtained ether during the course of aggregate execution
+    function drain() external payable {
+        payable(msg.sender).transfer(address(this).balance);
```

Comment:
We'll need to pass in the address to pay to since `msg.sender` would just be the same contract if attached as an aggregate call. We should also add a test where these are called in an `aggregate` call